playbooks/operationalizing/external-jenkins-integration.adoc

```diff
@@ -0,0 +1,339 @@
+---
+---
+= External Jenkins Integration
+Red Hat Containers & PaaS CoP
+v1.0, 2017-05-10
+:scripts_repo: https://github.com/rhtconsulting/rhc-ose
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+The OpenShift Container Platform provides a number of features to support Continuous Integration and Continuous Delivery. Aside from its native build and deployment capabilities, OpenShift contains tight integration with the link:Jenkins Continuous Integration server[https://jenkins.io/]. Not only can users deploy supported instances within the platform, but they can take advantage of such features such as first class support for the Jenkins "pipeline as code concept" as well as leveraging dynamically allocated slave instances for the execution of Jenkins jobs. A full list of features that are provided by the provided image can be found link:https://docs.openshift.com/container-platform/latest/using_images/other_images/jenkins.html[here]. While running the entire Jenkins ecosystem within OpenShift simplifies the configuration and management of a CI environment, many organizations have an existing Jenkins environment which they would like to make use of.
+
+This article describes the process for integrating an existing Jenkins instance located outside of the OpenShift cluster with OpenShift to be able to take advantage of many of the Jenkins related features provided by the platform.
+
+== Requirements
+
+The following are requirements in order to integrate an externally facing Jenkins instance into OpenShift:
+
+* Jenkins 2.x environment
+** User with administrator rights
+* OpenShift Container Platform 3.3+
+
+=== Networking
+
+To support connectivity between OpenShift and Jenkins, the following network level access must be available:
+
+* Jenkins must be able to communicate with the OpenShift API
+* OpenShift must be able to communicate with the following Jenkins ports
+** Web Port (8080 by default)
+** TCP port for JNLP Agents (50000 will be used below but can be any port of your choosing)
+
+== Plugins
+
+The majority of the capabilities provided by the integration between OpenShift and Jenkins are made possible through a collection of Jenkins plugins.
+
+* link:https://wiki.jenkins-ci.org/display/JENKINS/OpenShift+Pipeline+Plugin[OpenShift Pipeline] - The ability to trigger actions in OpenShift, such as starting a build or deployment. 
+* link:https://wiki.jenkins-ci.org/display/JENKINS/OpenShift+Sync+Plugin[OpenShift Sync] - Synchronizing the state of Jenkins jobs and OpenShift objects (BuildConfigs) in order to facilitate integrated pipelines. 
+* link:https://wiki.jenkins-ci.org/display/JENKINS/Kubernetes+Plugin[Kubernetes Plugin] - The ability to trigger actions in OpenShift, such as starting a build or deployment. 
```

Comment:
might also be worth pointing them to https://wiki.jenkins-ci.org/display/JENKINS/OpenShift+Client+Plugin